internal/ingest/ingest.go

```diff
@@ -135,14 +136,23 @@ func (ing *Ingester) Close() error {
 // gets to the last seen advertisement.  Then the entries in each advertisement
 // are synced and the multihashes in each entry are indexed.
 //
+// When the given selector is nil, a default selector is used where the
+// traversal stops once it reaches the currently latest synced advertisement.
+// Note that the reference to the latest synced advertisement returned by
+// GetLatestSync is only updated if no selector is given.  If the selector is
+// non-nil, the given selector is directly used as is for traversal; i.e. the
+// selector will not be wrapped with any stop logic.  For utility function to
+// construct such stop logic with custom selectors see: GetLatestSync,
+// legs.ExploreRecursiveWithStopNode.
+//
 // The Context argument controls the lifetime of the sync.  Canceling it
 // cancels the sync and causes the multihash channel to close without any data.
 //
 // Note that the multihash entries corresponding to the advertisement are
 // synced in the background.  The completion of advertisement sync does not
 // necessarily mean that the entries corresponding to the advertisement are
 // synced.
-func (ing *Ingester) Sync(ctx context.Context, peerID peer.ID, peerAddr multiaddr.Multiaddr) (<-chan multihash.Multihash, error) {
+func (ing *Ingester) Sync(ctx context.Context, peerID peer.ID, peerAddr multiaddr.Multiaddr, sel ipld.Node) (<-chan multihash.Multihash, error) {
```

Comment:
Do we really want to expose the ability to give any selector to sync?  While it is a more flexible API, I am wondering if it invites trouble.  If we think there may be use cases that need customized selectors, then we should keep this.  Otherwise, consider making the parameter being a simple depth limit and constructing the selector inside this function, in the interest of making an easier to use API.